server/src/main/java/org/candlepin/util/X509CRLEntryStream.java

```diff
@@ -0,0 +1,217 @@
+import java.io.ByteArrayOutputStream;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * Reads an X509 CRL in a stream and returns the serial number for a revoked certificate
+ * with each call to the iterator's next() function.
+ *
+ * The schema for an X509 CRL is described in
+ * <a href="https://tools.ietf.org/html/rfc5280#section-5">section 5 of RFC 5280</a>
+ *
+ * It is reproduced here for quick reference
+ *
+ * <pre>
+ * {@code
+ * CertificateList  ::=  SEQUENCE  {
+ *      tbsCertList          TBSCertList,
+ *      signatureAlgorithm   AlgorithmIdentifier,
+ *      signatureValue       BIT STRING  }
+ *
+ * TBSCertList  ::=  SEQUENCE  {
+ *      version                 Version OPTIONAL,
+ *                                   -- if present, MUST be v2
+ *      signature               AlgorithmIdentifier,
+ *      issuer                  Name,
+ *      thisUpdate              Time,
+ *      nextUpdate              Time OPTIONAL,
+ *      revokedCertificates     SEQUENCE OF SEQUENCE  {
+ *           userCertificate         CertificateSerialNumber,
+ *           revocationDate          Time,
+ *           crlEntryExtensions      Extensions OPTIONAL
+ *                                    -- if present, version MUST be v2
+ *                                }  OPTIONAL,
+ *      crlExtensions           [0]  EXPLICIT Extensions OPTIONAL
+ *                                    -- if present, version MUST be v2
+ *                                }
+ *
+ * Version, Time, CertificateSerialNumber, and Extensions
+ * are all defined in the ASN.1 in Section 4.1
+ *
+ * AlgorithmIdentifier is defined in Section 4.1.1.2
+ * }
+ * </pre>
+ *
+ * ASN1 is based around the TLV (tag, length, value) concept.  Any piece of
+ * data begins with a tag defining the data type, has a series of bytes to
+ * indicate the data length, and then the data itself.
+ *
+ * <b>This class does not perform any signature checking</b>.  Checking the signature is
+ * theoretically possible but we would be unable to check it until after the list of
+ * entries had been exhausted (since each entry would have to be sent through the hasher).  At
+ * that point, the user already has all the serial numbers.  The best approach would perhaps
+ * be to check the signature in hasNext() once revokedSeqBytes hits zero and throw a
+ * RuntimeException.  We would also need a RSA public key as a parameter to the constructor.
+ *
+ * See https://en.wikipedia.org/wiki/X.690 and http://luca.ntop.org/Teaching/Appunti/asn1.html
+ * for reference on ASN1 and DER encoding.
+ */
+public class X509CRLEntryStream implements Closeable, Iterator<X509CRLEntryObject> {
+    private InputStream crlStream;
+
+    // TODO should be a BigInteger?  Apparently long definite lengths can go up to 2^1008 - 1
```

Comment:
> Does this need to be done?

BouncyCastle's ASN1InputStream itself only uses integers for length.  MAXINT for Java is 2,147,483,647 which is roughly 2GB.  Right now the CRL list is at about 80MB so I don't think we'll need a BigInt there.  I can update the comment to indicate that.